### PR TITLE
CommutativeMonoid[Option[A]] from CommutativeSemigroup

### PR DESCRIPTION
**Pull request by valydia**
As mentioned by @kailuowang, this change will be binary breaking on scala 2.11, it is schedule for 2.1

### CODE REVIEW

**Comment by kailuowang** [comment]
Thanks, shall we also change these tests to CommutativeMonoid tests?
https://github.com/typelevel/cats/blob/master/kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala#L181-L182

File: kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala [hunk 1]
```diff
@@ -178,10 +178,8 @@ class Tests extends AnyFunSuiteLike with Discipline {
 
   checkAll("Monoid[String]", MonoidTests[String].monoid)
   checkAll("Monoid[String]", SerializableTests.serializable(Monoid[String]))
-  checkAll("Monoid[Option[Int]]", MonoidTests[Option[Int]].monoid)
-  checkAll("Monoid[Option[Int]]", SerializableTests.serializable(Monoid[String]))
   checkAll("Monoid[Option[String]]", MonoidTests[Option[String]].monoid)
-  checkAll("Monoid[Option[String]]", SerializableTests.serializable(Monoid[String]))
+  checkAll("Monoid[Option[String]]", SerializableTests.serializable(Monoid[Option[String]]))
```
Comment (valydia) [inline kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala]: It looked like a copy and paste went wrong

**Comment by valydia** [comment]
Yes, done there https://github.com/typelevel/cats/pull/2834/commits/cfd5742d1ad7b2f44b7ee6ba1479174fa44a1483

**Comment by LukaJCB** [comment]
Hi @valydia, can you rebase this with master? :)

File: kernel/src/main/scala/cats/kernel/instances/OptionInstances.scala [hunk 2]
```diff
@@ -80,3 +82,7 @@ class OptionMonoid[A](implicit A: Semigroup[A]) extends Monoid[Option[A]] {
         }
     }
 }
+
+class OptionCommutativeMonoid[A](implicit A: CommutativeSemigroup[A])
```
[the remark this answers is not in the document]
Comment (travisbrown) [reply kernel/src/main/scala/cats/kernel/instances/OptionInstances.scala]: @valydia Do you mind if I push this change and we can get this merged?

[the remark this answers is not in the document]
Comment (valydia) [reply kernel/src/main/scala/cats/kernel/instances/OptionInstances.scala]: Sure, go ahead! Sorry I've been busy lately! Thanks

**Comment by travisbrown** [comment]
@LukaJCB Think we can go ahead and merge this when green?

**Comment by LukaJCB** [comment]
Sounds good to me 👍

**Comment by valydia** [comment]
Thanks for the assist @travisbrown 👌

**Comment by kubukoz** [comment]
Can https://github.com/typelevel/cats/issues/2725 be closed now?

**Pull request merged**